### PR TITLE
fix(pf): remove __defaults entry from inputs

### LINKED ISSUE #2189: Importing PluginFramework based resources generates extraneous __defaults
### What happened?

The gory details are in https://github.com/pulumi/pulumi-aws/issues/4204

It looks like PF resources should not be injecting __defaults during import as they do not support it, but it might be happening through an unintended side-effect of code reuse. 

### Example

See above.

### Output of `pulumi about`

N/A

### Additional context

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by corymhall**
PF resources should not be injecting __defaults during
import as they do not support it, but it is happening through an
unintended side-effect of code reuse from sdkv2.

I could think of a couple other ways to handle this:

1. Duplicate the code in the pf and remove the code specific to
injecting __defaults
2. Removing where we inject an empty __defaults. See #2236

Decided to go with the simplest approach of just doing some post
processing since touching the `sdkv2` code seems to be a much bigger
effort.

fixes #2189, fixes pulumi/pulumi-aws#4204

### CODE REVIEW

File: pf/tfbridge/provider_read.go [hunk 1]
```diff
@@ -90,12 +90,35 @@ func (p *provider) ReadWithContext(
...
+		deleteDefaultsKey(&result.Inputs)
 	}
 
 	return result, ignoredStatus, err
 }
 
+// deleteDefaultsKey removes the `__defaults: []` entry from all objects recursively
+// The `__defaults` key is something used in sdkv2 and is not handled here in pf. Because
+// of some code reuse between sdkv2 & pf the `__defaults` key is getting inserted
+func deleteDefaultsKey(inputs *resource.PropertyMap) {
```
Comment (t0yv0) [inline pf/tfbridge/provider_read.go:102]: Have a look at propertyvalue.TransformPropertyValue that also goes under secrets etc and handles the recursion hopefully properly without any further bugs. Might be easier to review, we're just transforming everything and we're dropping __defaults whenever we find them.

Comment (corymhall) [reply pf/tfbridge/provider_read.go:103]: Done!

File: pf/tfbridge/provider_read.go [hunk 2]
```diff
@@ -90,12 +91,36 @@ func (p *provider) ReadWithContext(
...
+	i := *inputs
+	delete(i, "__defaults")
+	for _, key := range i.StableKeys() {
```
Comment (t0yv0) [inline pf/tfbridge/provider_read.go:106]: This is good. NIt: the for loop and top level delete may have been avoided by `propertyvalue.TransformPropertyValue` on `resource.NewObjectProperty(inputs)`. But this works too.

**Review by t0yv0 — APPROVED** [review]

**Review by iwahbe — APPROVED** [review]
1 nit, otherwise LGTM

File: pf/tfbridge/provider_read.go (hunk 2 above)
Comment (corymhall) [reply pf/tfbridge/provider_read.go]: It looks like `TransformPropertyValue` won't  run the `transform` function on the top level value, only on each value in the map downward.

**Pull request merged**